panel/chat/feed.py

```diff
@@ -681,6 +683,55 @@ def stream(
         self.param.trigger("_post_hook_trigger")
         return message
 
+    def create_steps(
+        self,
+        user: str | None = None,
+        avatar: str | bytes | BytesIO | None = None,
+        **steps_params
+    ) -> ChatSteps:
+        """
+        Creates a new ChatSteps component and streams it to the logs.
+
+        Arguments
+        ---------
+        user : str | None
+            The user to stream as; overrides the message's user if provided.
+            Will default to the user parameter.
+        avatar : str | bytes | BytesIO | None
+            The avatar to use; overrides the message's avatar if provided.
+            Will default to the avatar parameter.
+        steps_params : dict
+            Parameters to pass to the ChatSteps.
+
+        Returns
+        -------
+        The ChatSteps that was created.
+        """
+        steps = ChatSteps(**steps_params)
+        self.stream(steps, user=user, avatar=avatar)
+        return steps
+
+    def attach_step(self, objects: str | list[str] | None = None, **step_params) -> ChatStep:
```

Comment:
What happens right now if you `.stream()` when the last message contains `ChatSteps` could that not simply be used to attach a new step?

Reply:
If this can't be removed I'd suggest renaming maybe, maybe `add_step` or `stream_step`?

Reply:
`append_step`?

Reply:
I decided against adding the functionality to `stream` since it can get quite complex due to the kwargs `value` accepting `ChatMessage` types and also `message`.

Instead, I compressed the two functions into `append_step`, where it accepts a `steps` kwarg which can be `new`, `append`, or a `ChatSteps` type.